build.c

```diff
@@ -106,6 +106,10 @@ queue(struct edge *e)
 {
 	struct edge **front = &work;
 
+	if (e->flags & FLAG_QUEUED)
+		return;
+	e->flags |= FLAG_QUEUED;
```

Comment:
Could you explain why we would otherwise queue edges multiple times with dyndeps?